codex-rs/core/src/config.rs

```diff
@@ -214,6 +225,33 @@ fn load_config_as_toml(codex_home: &Path) -> std::io::Result<TomlValue> {
     }
 }
 
+/// Patch `CODEX_HOME/config.toml` project state.
+/// Use with caution.
+pub fn set_project_trusted(
+    codex_home: &Path,
+    project_path: &Path,
+    trusted: bool,
+) -> anyhow::Result<()> {
+    let config_path = codex_home.join("config.toml");
+
+    // Parse existing config if present; otherwise start a new document.
+    let mut doc = match std::fs::read_to_string(&config_path) {
+        Ok(s) => s.parse::<DocumentMut>()?,
+        Err(e) if e.kind() == std::io::ErrorKind::NotFound => DocumentMut::new(),
+        Err(e) => return Err(e.into()),
+    };
+
+    let project_key = project_path.to_string_lossy().to_string();
+    doc["projects"][project_key.as_str()]["trusted"] = toml_edit::value(trusted);
```

Comment:
What happens if `doc` does not have a `"projects"` key: does it create an entry on demand? How does this not panic?